core/inject.js

```diff
@@ -329,6 +189,140 @@ Blockly.createMainWorkspace_ = function(svg, options, blockDragSurface,
   return mainWorkspace;
 };
 
+/**
+ * Extracts the object from the given event.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace the event originated
+ *    from.
+ * @param {!Blockly.Events.BumpEvent} e An event containing an object.
+ * @return {?Blockly.BlockSvg|?Blockly.WorkspaceCommentSvg} The extracted
+ *    object.
+ * @private
+ */
+Blockly.extractObjectFromEvent_ = function(workspace, e) {
+  var object = null;
+  switch (e.type) {
+    case Blockly.Events.BLOCK_CREATE:
+    case Blockly.Events.BLOCK_MOVE:
+      object = workspace.getBlockById(e.blockId);
+      if (object) {
+        object = object.getRootBlock();
+      }
+      break;
+    case Blockly.Events.COMMENT_CREATE:
+    case Blockly.Events.COMMENT_MOVE:
+      object = workspace.getCommentById(e.commentId);
+      break;
+  }
+  return object;
+};
+
+/**
+ * Bumps the top objects in the given workspace into bounds.
+ * @param {!Blockly.WorkspaceSvg} workspace The workspace.
```

Comment:
Do these methods need to be public?

Reply:
No, I think you're right. private or protected would be more appropriate (protected if it ends up being refactored out).